cmd/agent/common/common.go

```diff
@@ -82,10 +83,49 @@ func NewRemoteConfigService(hostname string) (*remoteconfig.Service, error) {
 	baseRawURL := configUtils.GetMainEndpoint(config.Datadog, "https://config.", "remote_configuration.rc_dd_url")
 	traceAgentEnv := configUtils.GetTraceAgentDefaultEnv(config.Datadog)
 
-	configService, err := remoteconfig.NewService(config.Datadog, apiKey, baseRawURL, hostname, remoteconfig.WithTraceAgentEnv(traceAgentEnv))
+	telemetryReporter := newRcTelemetryReporter()
+
+	configService, err := remoteconfig.NewService(config.Datadog, apiKey, baseRawURL, hostname, telemetryReporter, remoteconfig.WithTraceAgentEnv(traceAgentEnv))
 	if err != nil {
 		return nil, fmt.Errorf("unable to create remote-config service: %w", err)
 	}
 
 	return configService, nil
 }
+
+// ddRcTelemetryReporter is a datadog-agent telemetry counter for RC cache bypass metrics. It implements the RcTelemetryReporter interface.
+type ddRcTelemetryReporter struct {
```

Comment:
Can you move all this code (+ the `NewRemoteConfigService` function above) in a separate file owned by RC ?

Reply:
I think this makes sense! I will likely follow this approach, but want to make sure it satisfies @hush-hush 's comment as well 👍 🙇